From d6c265e5bb8e811b8f0a149c34e948304072a0f7 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 14 Feb 2024 11:30:22 +0100
Subject: GCC: template specialization out of namespace in BodyConsumerBase

GCC does not allow template specializations in class namespace, move
it to blink namespace.

Bug: 40565911
Change-Id: Id5e9cb2c526690777a4ca014a34e4a9466911946
---
 third_party/blink/renderer/core/fetch/body.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/third_party/blink/renderer/core/fetch/body.cc b/third_party/blink/renderer/core/fetch/body.cc
index 9b802e592aad2..b13896898e4f5 100644
--- a/third_party/blink/renderer/core/fetch/body.cc
+++ b/third_party/blink/renderer/core/fetch/body.cc
@@ -94,6 +94,11 @@ class BodyConsumerBase : public GarbageCollected<BodyConsumerBase>,
   const scoped_refptr<base::SingleThreadTaskRunner> task_runner_;
 };
 
+template <>
+void BodyConsumerBase::ResolveNow(const Persistent<DisallowNewWrapper<ScriptValue>>& object) {
+  resolver_->Resolve(object->Value());
+}
+
 class BodyBlobConsumer final : public BodyConsumerBase {
  public:
   using BodyConsumerBase::BodyConsumerBase;
-- 
2.40.1

